util/chplenv/chpl_jemalloc.py

```diff
@@ -22,6 +29,17 @@ def get():
     if mem_val != 'jemalloc' and jemalloc_val != 'none':
       error("CHPL_JEMALLOC must be 'none' when CHPL_MEM is not jemalloc")
 
+    if flag == 'host':
+      platform_val = chpl_platform.get('target')
+      linux = platform_val.startswith('linux')
+      darwin = platform_val.startswith('darwin')
+
+      if linux and mem_val == 'jemalloc' and jemalloc_val == 'system':
+        error("CHPL_JEMALLOC=system is not supported on Linux for host builds")
+
+      if darwin and mem_val == 'jemalloc' and jemalloc_val == 'bundled':
+        error("CHPL_JEMALLOC=bundled is not supported on Mac for host ubilds")
```

Comment:
This is probably captured in an issue, but what prevents us from using the bundled jemalloc on mac?  Is it just that we're only building the static library and we need a dylib on mac for things to work correctly? If so we could specialize the host build for mac to produce the dynamic lib instead of building the static lib only.

Reply:
Nothing preventing us. That was my call after running into some other bundled shared library weirdness (see [this comment](https://github.com/Cray/chapel-private/issues/2629#issuecomment-952324222)). I can add it and it would work as well as the other shared libraries I think. (Also I typo'd `ubilds`)